src/files/manager.ts

```diff
@@ -229,4 +235,56 @@ export class FilesManager {
       throw error;
     }
   }
+
+  /**
+   * Uploads a new media file to the Strapi Media Library.
+   *
+   * @param file - The file to upload (Blob).
+   * @param filename - Optional filename to use for the uploaded file.
+   * @returns A promise that resolves to the uploaded file information.
+   *
+   * @throws {FileForbiddenError} if the user does not have permission to upload files.
+   * @throws {HTTPError} if the HTTP client encounters connection issues, the server is unreachable, or authentication fails.
+   *
+   * @example
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * // Upload with a File object (browser)
+   * const fileInput = document.querySelector('input[type="file"]');
+   * const file = fileInput.files[0];
+   * const result = await filesManager.upload(file);
+   *
+   * // Upload with a Blob and custom filename (Node.js)
+   * import { blobFrom } from 'node-fetch';
+   * const file = await blobFrom('./1.png', 'image/png');
+   * const result = await filesManager.upload(file);
+   *
+   * console.log(result); // Upload response with file details
+   * ```
+   */
+  async upload(file: Blob, filename?: string): Promise<MediaUploadResponse> {
```

Comment:
```
  async upload(file: Blob | Buffer, filename?: string): Promise<MediaUploadResponse> {
```
I'm not sure if this is automatically compatible though, might need to check if it's a Buffer and do something slightly different

Reply:
```
async upload(file: Blob, filename?: string, info?: FileUpdateData): Promise<MediaUploadResponse> {
```
Is this possible? It seems weird to have to upload in two steps to add the metadata